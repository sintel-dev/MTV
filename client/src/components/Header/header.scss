@import '../../assets/sass/colors.scss';
@import '../../assets/sass/main.scss';
.main-header {
    box-shadow: 0 0 14px #000;
    background: $main-color;
    position: fixed;
    left: 0;
    right: 0;
    top: 0;
    z-index: 20;

    .header-container {
        width: 96%;
        margin: 0 auto;
    }

    a {
        transition: all 0.2s ease-in-out;
        color: #fff;
        &.logo {
            background: $main-color;
            text-align: left;
            font-size: 28px;
            line-height: 50px;
            padding: 0;
            height: 50px;
            float: left;
            width: 73px;
            font-weight: 300;
            overflow: hidden;
            font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;
            transition: width 0.3s ease-in-out;
            text-decoration: none;
        }

        &:hover,
        &:focus {
            text-decoration: none;
        }
    }

    .page-switch-btn {
        float: left;
        background-color: transparent;
        background-image: none;
        padding: 15px;
        display: none;

        &.active {
            display: inline-block;
        }
    }

    .user-opts{
        float: right;
        line-height: 40px;
        a{
            text-decoration: none;
            &:hover{
                text-decoration: underline;
            }
        }
    }

    .logout-button{
        background-color: transparent;
        font-size: 16px;
        border: none;
        color: #fff;
        cursor: pointer;
        outline: none;
    }

    .exp-info{
        float: left;
        font-family: 'Roboto';

        ul{
            width: 200px;
            display: inline-block;
            font-size: 16px;
            border: 1px solid rgba(151,151,151,0.5);
            border-radius: 4px;
            box-shadow: 0 2px 4px -1px rgba(0,0,0,0.2), 0 4px 5px 0 rgba(0,0,0,0.14), 0 1px 10px 0 rgba(0,0,0,0.12);
            padding: 5px 15px;
            margin: 8px 0 0 50px;
            cursor: pointer;
            font-weight: bold;
            background: #282A30;

            &:hover{
                span{
                    path{
                        fill: #fff;
                    }
                }
            }

            li{
                position: relative;

                span{
                    float: right;

                    path{
                        fill: #979797;
                        transition: all .2s ease-in-out;
                    }

                    &:last-child{
                        font-weight: normal;
                    }
                }

                ul{
                    border-top-left-radius: 0;
                    border-top-right-radius: 0;
                    border-top: none;
                    position: absolute;
                    left: -16px;
                    margin: 4px 0 0 0;
                    padding: 10px 15px;
                    transition: all .2s ease-in-out;
                    top: 21px;
                    display: none;
                }

                &.active{
                    border-bottom: none;
                    box-shadow: none;

                    ul{
                        display: block;
                    }

                    span{
                        path{
                            fill: #fff;
                        }
                    }
                }

                li{
                    display: flex;
                    cursor: initial;
                    font-size: 14px;
                    text-transform: none;
                    margin-bottom: 8px;

                    span{
                        flex: 1;
                    }
                }
            }
        }

    }
}
