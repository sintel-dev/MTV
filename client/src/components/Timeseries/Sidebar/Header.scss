@import '../../../assets/sass/form';

.sidebar-heading{
    ul{
        float: none;
    }

    li{
        margin-right: 10px;
        vertical-align: middle;
        &.signal-title{
            font-size: 20px;
        }
    }

    .switch-control{
        font-size: 14px;
    }
}
