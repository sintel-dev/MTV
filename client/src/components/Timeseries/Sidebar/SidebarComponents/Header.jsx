import React, { Component } from 'react';
import { connect } from 'react-redux';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faChevronRight } from '@fortawesome/free-solid-svg-icons';
import EventSummary from './EventSummary';
import {
  getDatarunDetails,
  getSelectedPeriodLevel,
  getIsEditingEventRange,
  getGrouppedDatarunEvents,
  getIsEventModeEnabled,
  getIsTimeSyncModeEnabled,
  getFilteredPeriodRange,
  getScrollHistory,
} from '../../../../model/selectors/datarun';
import {
  toggleEventModeAction,
  setPeriodRangeAction,
  setScrollHistoryAction,
  setReviewPeriodAction,
} from '../../../../model/actions/datarun';
import { toggleRelativeScale } from 'src/model/actions/sidebar';
import { getIsRelativeScaleEnabled } from 'src/model/selectors/sidebar';

class Header extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isSummaryVisible: true,
      // isRelativeScaleEnabled: false,
    };
    this.toggleSummaryDetails = this.toggleSummaryDetails.bind(this);
  }

  componentDidUpdate(prevProps) {
    const { isTimeSyncEnabled, filteredPeriodRange, setScrollHistory, scrollHistory } = this.props;

    if (!isTimeSyncEnabled) {
      return;
    }

    if (prevProps.filteredPeriodRange[0].level !== filteredPeriodRange[0].level) {
      const { level } = filteredPeriodRange[0];
      let currentPeriod = scrollHistory;
      if (level === 'year') {
        currentPeriod = {
          ...currentPeriod,
          level: 'year',
        };
      }

      if (level === 'month') {
        const currentYear = currentPeriod.year;
        const newYear = filteredPeriodRange[0].parent.name;
        currentPeriod = {
          ...currentPeriod,
          year: currentYear !== newYear ? newYear : currentYear,
          level: 'month',
        };
      }

      if (level === 'day') {
        const currentMonth = currentPeriod.month;
        const newMonth = filteredPeriodRange[0].parent.name;
        const currentYear = currentPeriod.year;
        const newYear = filteredPeriodRange[0].parent.parent.name;
        currentPeriod = {
          ...currentPeriod,
          year: currentYear !== newYear ? newYear : currentYear,
          month: currentMonth !== newMonth ? newMonth : currentMonth,
          level: 'day',
        };
      }
      setScrollHistory(currentPeriod);
    }
  }

  toggleSummaryDetails() {
    const { isSummaryVisible } = this.state;
    this.setState({
      isSummaryVisible: !isSummaryVisible,
    });
  }

  renderHeadingControls() {
    const buttonText = this.state.isSummaryVisible ? 'HIDE' : 'SHOW';
    return (
      <div className="sidebar-heading">
        <ul>
          <li className="signal-title">Periodical View</li>
          <li>
            <button type="button" onClick={this.toggleSummaryDetails} id="toggleSummary">
              <span>{buttonText}</span>
              <FontAwesomeIcon icon={faChevronRight} />
            </button>
          </li>
        </ul>
      </div>
    );
  }

  showPeriod() {
    const { filteredPeriodRange, selectedPeriodLevel, isTimeSyncEnabled } = this.props;
    let periodString = 'YY/MM';

    if (isTimeSyncEnabled) {
      const { level } = filteredPeriodRange[0];
      if (level === 'month') {
        periodString = periodString.replace('YY', filteredPeriodRange[0].parent.name);
      }

      if (level === 'day') {
        periodString = periodString.replace('YY', filteredPeriodRange[0].parent.parent.name);
        periodString = periodString.replace('MM', filteredPeriodRange[0].parent.name);
      }
    } else {
      if (selectedPeriodLevel.year) {
        periodString = periodString.replace('YY', selectedPeriodLevel.year);
      }
      if (selectedPeriodLevel.month) {
        periodString = periodString.replace('MM', selectedPeriodLevel.month);
      }
    }

    return (
      <div className="period-info">
        <p>{periodString}</p>
      </div>
    );
  }

  // changeScale() {
  //   const { isRelativeScaleEnabled } = this.state;
  //   this.setState({
  //     isRelativeScaleEnabled: !isRelativeScaleEnabled,
  //   });
  // }

  render() {
    const { isSummaryVisible } = this.state;
    const {
      setReviewPeriod,
      selectedPeriodLevel,
      isEditingEventRange,
      grouppedEvents,
      filteredPeriodRange,
      currentPeriod,
      scrollHistory,
      isTimeSyncEnabled,
      isEventModeEnabled,
      toggleEventsMode,
      isRelativeScaleEnabled,
      toggleRelativeScale,
      dataRun,
    } = this.props;

    const getBtnProps = (button) => {
      const getParentLevel = () => (button === 'day' ? 'month' : 'year');
      const getIsActive = () => {
        if (button === 'year') {
          return isTimeSyncEnabled ? scrollHistory.level === 'year' : currentPeriod.level === null;
        }
        return isTimeSyncEnabled
          ? scrollHistory.level === button
          : currentPeriod.level === getParentLevel() || currentPeriod.level === getParentLevel();
      };

      const getIsDisabled = () => {
        if (button === 'year') {
          return false;
        }
        return isTimeSyncEnabled ? scrollHistory[getParentLevel()] === null : currentPeriod[getParentLevel()] === null;
      };

      return {
        className: getIsActive() ? 'active' : '',
        disabled: getIsDisabled(),
        onClick: () =>
          !isEditingEventRange && button === 'year' ? setReviewPeriod(null) : setReviewPeriod(getParentLevel()),
      };
    };

    return (
      <div className="period-control">
        {/* @TODO - under discussion if this should be removed or not */}
        {/* {this.renderHeadingControls()} */}
        <EventSummary
          selectedPeriodLevel={selectedPeriodLevel}
          grouppedEvents={grouppedEvents}
          filteredPeriodRange={filteredPeriodRange}
          signalName={dataRun.signal}
          isOpen={isSummaryVisible}
          isTimeSyncEnabled={isTimeSyncEnabled}
        />
        <div className="period-wrapper">
          <div className="sidechart-controls switch-control">
            <div className="row">
              <label htmlFor="glyphScale">
                <input
                  type="checkbox"
                  id="glyphScale"
                  onChange={() => toggleRelativeScale()}
                  checked={isRelativeScaleEnabled}
                />
                <span className="switch" />
                Relative scale
              </label>
            </div>
          </div>
          <ul className="period-filter">
            <li>
              <button type="button" {...getBtnProps('year')}>
                Year
              </button>
            </li>
            <li>
              <button type="button" {...getBtnProps('month')}>
                Month
              </button>
            </li>
            <li>
              <button type="button" {...getBtnProps('day')}>
                Day
              </button>
            </li>
          </ul>
          {/* @TODO - new toggle switch appear(changeScale), check to see where to place this one */}
          {/* <ul>
            <li>
              <div className="switch-control reversed">
                <div className="row">
                  <label htmlFor="toggleEvents">
                    Show Events
                    <input
                      type="checkbox"
                      id="toggleEvents"
                      onChange={(event) => toggleEventsMode(event.target.checked)}
                      checked={isEventModeEnabled}
                    />
                    <span className="switch" />
                  </label>
                </div>
              </div>
            </li>
          </ul> */}
          {/* {this.showPeriod(selectedPeriodLevel)} */}
        </div>
        <div className="clear" />
      </div>
    );
  }
}

export default connect(
  (state) => ({
    dataRun: getDatarunDetails(state),
    selectedPeriodLevel: getSelectedPeriodLevel(state),
    isEditingEventRange: getIsEditingEventRange(state),
    grouppedEvents: getGrouppedDatarunEvents(state),
    isEventModeEnabled: getIsEventModeEnabled(state),
    isTimeSyncEnabled: getIsTimeSyncModeEnabled(state),
    filteredPeriodRange: getFilteredPeriodRange(state),
    currentPeriod: getSelectedPeriodLevel(state),
    scrollHistory: getScrollHistory(state),
    isRelativeScaleEnabled: getIsRelativeScaleEnabled(state),
  }),
  (dispatch) => ({
    setPeriodRange: (periodRange) => dispatch(setPeriodRangeAction(periodRange)),
    toggleEventsMode: (mode) => dispatch(toggleEventModeAction(mode)),
    setScrollHistory: (period) => dispatch(setScrollHistoryAction(period)),
    setReviewPeriod: (period) => dispatch(setReviewPeriodAction(period)),
    toggleRelativeScale: () => dispatch(toggleRelativeScale()),
  }),
)(Header);
