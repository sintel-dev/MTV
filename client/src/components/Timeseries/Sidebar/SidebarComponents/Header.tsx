import React, { Component, ReactNode } from 'react';
import { connect } from 'react-redux';
import { toggleRelativeScaleAction } from 'src/model/actions/sidebar';
import { getIsRelativeScaleEnabled } from 'src/model/selectors/sidebar';
import {
  getSelectedPeriodLevel,
  getIsEditingEventRange,
  getIsEventModeEnabled,
  getIsTimeSyncModeEnabled,
  getFilteredPeriodRange,
  getScrollHistory,
} from 'src/model/selectors/datarun';
import {
  toggleEventModeAction,
  setPeriodRangeAction,
  setScrollHistoryAction,
  setReviewPeriodAction,
} from 'src/model/actions/datarun';
import { RootState } from 'src/model/types';
import EventSummary from './EventSummary';

const showPeriod = (periodRange): ReactNode => {
  let periodString = (
    <p>
      <span>YY / MM</span>
    </p>
  );

  const { level } = periodRange;
  if (level === 'month') {
    periodString = (
      <p>
        <span className="active">{periodRange.parent.name}</span>
        <span> / MM</span>
      </p>
    );
  }

  if (level === 'day') {
    periodString = (
      <p>
        <span className="active">{periodRange.parent.parent.name}</span> /{' '}
        <span className="active">{periodRange.parent.name}</span>
      </p>
    );
  }

  return <div className="period-info">{periodString}</div>;
};

type StateProps = ReturnType<typeof mapState>;
type DispatchProps = ReturnType<typeof mapDispatch>;
type Props = StateProps & DispatchProps;

class Header extends Component<Props> {
  componentDidUpdate(prevProps: Props) {
    const { isTimeSyncEnabled, filteredPeriodRange, setScrollHistory, scrollHistory } = this.props;

    if (!isTimeSyncEnabled) {
      return;
    }

    if (prevProps.filteredPeriodRange[0].level !== filteredPeriodRange[0].level) {
      const { level } = filteredPeriodRange[0];
      let currentPeriod = scrollHistory;
      if (level === 'year') {
        currentPeriod = {
          ...currentPeriod,
          level: 'year',
        };
      }

      if (level === 'month') {
        const currentYear = currentPeriod.year;
        const newYear = filteredPeriodRange[0].parent.name;
        currentPeriod = {
          ...currentPeriod,
          year: currentYear !== newYear ? newYear : currentYear,
          level: 'month',
        };
      }

      if (level === 'day') {
        const currentMonth = currentPeriod.month;
        const newMonth = filteredPeriodRange[0].parent.name;
        const currentYear = currentPeriod.year;
        const newYear = filteredPeriodRange[0].parent.parent.name;
        currentPeriod = {
          ...currentPeriod,
          year: currentYear !== newYear ? newYear : currentYear,
          month: currentMonth !== newMonth ? newMonth : currentMonth,
          level: 'day',
        };
      }
      setScrollHistory(currentPeriod);
    }
  }

  private getBtnProps(button: string): { className: string; disabled: boolean; onClick: () => void } {
    const { setReviewPeriod, isEditingEventRange, currentPeriod, scrollHistory, isTimeSyncEnabled } = this.props;
    const getParentLevel = (): string => (button === 'day' ? 'month' : 'year');
    const getIsActive = (): string | boolean | null => {
      if (button === 'year') {
        return isTimeSyncEnabled ? scrollHistory.level === 'year' : currentPeriod.level === null;
      }
      return isTimeSyncEnabled
        ? scrollHistory.level === button
        : currentPeriod.level === getParentLevel() || currentPeriod.level === getParentLevel();
    };

    const getIsDisabled = (): boolean => {
      if (button === 'year') {
        return false;
      }
      return isTimeSyncEnabled ? scrollHistory[getParentLevel()] === null : currentPeriod[getParentLevel()] === null;
    };

    return {
      className: getIsActive() ? 'active' : '',
      disabled: getIsDisabled(),
      onClick: () =>
        !isEditingEventRange && button === 'year' ? setReviewPeriod(null) : setReviewPeriod(getParentLevel()),
    };
  }

  render() {
    const { filteredPeriodRange, isRelativeScaleEnabled, toggleRelativeScale } = this.props;

    return (
      <div className="period-control">
        <EventSummary />
        <div className="period-wrapper">
          <div className="sidechart-controls switch-control">
            <div className="row">
              <label htmlFor="glyphScale">
                <input
                  type="checkbox"
                  id="glyphScale"
                  onChange={toggleRelativeScale}
                  checked={isRelativeScaleEnabled}
                />
                <span className="switch" />
                Relative scale
              </label>
            </div>
          </div>
          {showPeriod(filteredPeriodRange[0])}
          <ul className="period-filter">
            <li>
              <button type="button" {...this.getBtnProps('year')}>
                Year
              </button>
            </li>
            <li>
              <button type="button" {...this.getBtnProps('month')}>
                Month
              </button>
            </li>
            <li>
              <button type="button" {...this.getBtnProps('day')}>
                Day
              </button>
            </li>
          </ul>
          {/* @TODO - new toggle switch appear(changeScale), check to see where to place this one */}
          {/* <ul>
            <li>
              <div className="switch-control reversed">
                <div className="row">
                  <label htmlFor="toggleEvents">
                    Show Events
                    <input
                      type="checkbox"
                      id="toggleEvents"
                      onChange={(event) => toggleEventsMode(event.target.checked)}
                      checked={isEventModeEnabled}
                    />
                    <span className="switch" />
                  </label>
                </div>
              </div>
            </li>
          </ul> */}
        </div>
        <div className="clear" />
      </div>
    );
  }
}

const mapState = (state: RootState) => ({
  isEditingEventRange: getIsEditingEventRange(state),
  isEventModeEnabled: getIsEventModeEnabled(state),
  isTimeSyncEnabled: getIsTimeSyncModeEnabled(state),
  filteredPeriodRange: getFilteredPeriodRange(state),
  currentPeriod: getSelectedPeriodLevel(state),
  scrollHistory: getScrollHistory(state),
  isRelativeScaleEnabled: getIsRelativeScaleEnabled(state),
});

const mapDispatch = (dispatch: Function) => ({
  setPeriodRange: (periodRange) => dispatch(setPeriodRangeAction(periodRange)),
  toggleEventsMode: (mode) => dispatch(toggleEventModeAction(mode)),
  setScrollHistory: (period) => dispatch(setScrollHistoryAction(period)),
  setReviewPeriod: (period) => dispatch(setReviewPeriodAction(period)),
  toggleRelativeScale: () => dispatch(toggleRelativeScaleAction()),
});

export default connect<StateProps, DispatchProps, {}, RootState>(mapState, mapDispatch)(Header);
