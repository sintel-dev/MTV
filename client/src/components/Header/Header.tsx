import React, { useState } from 'react';
import './header.scss';
import { connect } from 'react-redux';
import { Link, useLocation } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faChevronRight, faChevronLeft, faCaretDown } from '@fortawesome/free-solid-svg-icons';
import { getCurrentExperimentDetails } from 'src/model/selectors/experiment';
import { getSelectedExperiment } from '../../model/selectors/projects';
import { onUserLogoutAction } from '../../model/actions/users';
import { RootState } from '../../model/types';

type StateProps = ReturnType<typeof mapState>;
type DispatchProps = ReturnType<typeof mapDispatch>;
type Props = StateProps & DispatchProps;

export const Header: React.FC<Props> = (props) => {
  const isSwitchVisible = props.selectedExperimentID ? 'active' : '';
  const { experimentDetails } = props;
  let location = useLocation();

  const currentView = location.pathname;
  const linkTo = currentView === '/' ? `/experiment/${props.selectedExperimentID}` : '/';

  const logoutUser = () => {
    props.userLogout();
    window.location.href = '/';
  };

  const [isInfoOpen, toggleInfo] = useState(false);
  const activeClass = isInfoOpen ? 'active' : '';

  window.addEventListener('click', (evt: any) => {
    if (currentView === '/') {
      return null;
    }
    const dropdown = document.querySelector('.exp-info');
    dropdown && !dropdown.contains(evt.target) && toggleInfo(false);
  });

  return (
    <header id="header" className="main-header">
      <div className="header-container">
        <Link to="/" className="logo">
          <b>MTV</b>
        </Link>
        <Link to={linkTo} className={`page-switch-btn ${isSwitchVisible}`}>
          <FontAwesomeIcon icon={currentView === '/' ? faChevronRight : faChevronLeft} />
        </Link>
        {currentView !== '/' && (
          <div className="exp-info" onClick={() => toggleInfo(!isInfoOpen)}>
            <ul>
              <li className={activeClass}>
                Details
                <span>
                  <FontAwesomeIcon icon={faCaretDown} />
                </span>
                <ul>
                  <li>
                    <span>Pipeline:</span>
                    <span>{experimentDetails.pipeline}</span>
                  </li>
                  <li>
                    <span>Dataset:</span>
                    <span>{experimentDetails.dataset}</span>
                  </li>
                  <li>
                    <span>By:</span>
                    <span>{experimentDetails.created_by || 'Unknown'}</span>
                  </li>
                  <li>
                    <span>Project:</span>
                    <span>{experimentDetails.project}</span>
                  </li>
                </ul>
              </li>
            </ul>
          </div>
        )}

        <ul className="user-opts">
          <li>
            <button type="button" onClick={logoutUser} className="logout-button">
              Logout
            </button>
          </li>
        </ul>
      </div>
    </header>
  );
};

const mapState = (state: RootState) => ({
  selectedExperimentID: getSelectedExperiment(state),
  experimentDetails: getCurrentExperimentDetails(state),
});

const mapDispatch = (dispatch: Function) => ({
  userLogout: () => dispatch(onUserLogoutAction()),
});

export default connect<StateProps, DispatchProps, {}, RootState>(mapState, mapDispatch)(Header);
