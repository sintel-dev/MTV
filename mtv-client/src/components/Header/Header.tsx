import React from 'react';
import './header.scss';

const Header: React.FC = () => (
  <header id="header" className="main-header">
    <a href="/" className="logo"><b>MTV</b></a>
    <a href="/" className="page-switch-btn">Switch</a>
  </header>);

export default Header;
