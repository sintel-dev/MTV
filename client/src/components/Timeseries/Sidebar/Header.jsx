import React from 'react';
import { connect } from 'react-redux';
import EventSummary from './EventSummary';
import {
  getDatarunDetails,
  getReviewPeriod,
  getSelectedPeriodLevel,
  getIsEditingEventRange,
  getGrouppedDatarunEvents,
  getIsEventModeEnabled,
  getIsTimeSyncModeEnabled,
} from '../../../model/selectors/datarun';
import { reviewPeriodAction, toggleEventModeAction, toggleTimeSyncModeAction } from '../../../model/actions/datarun';
import './Header.scss';

const showPeriod = (selectedPeriodLevel) => {
  let periodString = 'YY/MM';
  if (selectedPeriodLevel.year) {
    periodString = periodString.replace('YY', selectedPeriodLevel.year);
  }
  if (selectedPeriodLevel.month) {
    periodString = periodString.replace('MM', selectedPeriodLevel.month);
  }

  return (
    <div className="period-info">
      <p>{periodString}</p>
    </div>
  );
};

const SidebarHeading = ({ signalName, toggleEvent, isEventModeEnabled, toggleTimeSync, isTimeSyncEnabled }) => (
  <div className="sidebar-heading">
    <ul>
      <li className="signal-title">{signalName}</li>
      <li>
        <div className="switch-control">
          <div className="row">
            <label htmlFor="toggleEvents">
              <input
                type="checkbox"
                id="toggleEvents"
                onChange={(event) => toggleEvent(event.target.checked)}
                checked={isEventModeEnabled}
              />
              <span className="switch" />
              Show Events
            </label>
          </div>
        </div>
      </li>
      <li>
        <div className="switch-control">
          <div className="row">
            <label htmlFor="toggleTimeSync">
              <input
                type="checkbox"
                id="toggleTimeSync"
                onChange={(event) => toggleTimeSync(event.target.checked)}
                checked={isTimeSyncEnabled}
              />
              <span className="switch" />
              Sync Time Ranges
            </label>
          </div>
        </div>
      </li>
    </ul>
  </div>
);

const Header = ({
  dataRun,
  setReviewRange,
  reviewRange,
  selectedPeriodLevel,
  isEditingEventRange,
  grouppedEvents,
  toggleEventsMode,
  isEventModeEnabled,
  toggleTimeSync,
  isTimeSyncEnabled,
}) => (
  <div className="period-control">
    <SidebarHeading
      signalName={dataRun.signal}
      toggleEvent={toggleEventsMode}
      isEventModeEnabled={isEventModeEnabled}
      toggleTimeSync={toggleTimeSync}
      isTimeSyncEnabled={isTimeSyncEnabled}
    />
    <EventSummary selectedPeriodLevel={selectedPeriodLevel} grouppedEvents={grouppedEvents} />
    <div>
      {showPeriod(selectedPeriodLevel)}
      <ul className="period-filter">
        <li>
          <button
            type="button"
            disabled={isTimeSyncEnabled}
            onClick={() => !isEditingEventRange && setReviewRange('year')}
            className={reviewRange === 'year' || reviewRange === null ? 'active' : ''}
          >
            Year
          </button>
        </li>
        <li>
          <button
            type="button"
            onClick={() => !isEditingEventRange && setReviewRange('month')}
            className={reviewRange === 'month' ? 'active' : ''}
            disabled={reviewRange === null || isTimeSyncEnabled}
          >
            Month
          </button>
        </li>
        <li>
          <button
            type="button"
            onClick={() => !isEditingEventRange && setReviewRange('day')}
            className={reviewRange === 'day' ? 'active' : ''}
            disabled={reviewRange === null || selectedPeriodLevel.month === '' || isTimeSyncEnabled}
          >
            Day
          </button>
        </li>
      </ul>
    </div>
    <div className="clear" />
  </div>
);

export default connect(
  (state) => ({
    dataRun: getDatarunDetails(state),
    reviewRange: getReviewPeriod(state),
    selectedPeriodLevel: getSelectedPeriodLevel(state),
    isEditingEventRange: getIsEditingEventRange(state),
    grouppedEvents: getGrouppedDatarunEvents(state),
    isEventModeEnabled: getIsEventModeEnabled(state),
    isTimeSyncEnabled: getIsTimeSyncModeEnabled(state),
  }),
  (dispatch) => ({
    setReviewRange: (periodLevel) => dispatch(reviewPeriodAction(periodLevel)),
    toggleEventsMode: (mode) => dispatch(toggleEventModeAction(mode)),
    toggleTimeSync: (mode) => dispatch(toggleTimeSyncModeAction(mode)),
  }),
)(Header);
