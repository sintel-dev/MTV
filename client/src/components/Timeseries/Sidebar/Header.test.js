import React from 'react';
import { Header } from './Header';
import { dataRun } from '../../../tests/testmocks/dataRun';

jest.mock('./EventSummary', () => () => 'Event Summary component');

describe('Testing Sidebar header component', () => {
  const headerProps = {
    dataRun,
    selectedPeriodLevel: {},
  };

  it('Should render without crashing', () => {
    const header = shallow(<Header {...headerProps} />);
    expect(header).toMatchSnapshot();
  });
});
