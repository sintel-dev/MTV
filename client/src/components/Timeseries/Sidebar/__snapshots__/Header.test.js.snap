// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Sidebar header component Should render without crashing 1`] = `
<div
  className="period-control"
>
  <SidebarHeading
    signalName="S-1"
  />
  <Component
    selectedPeriodLevel={Object {}}
  />
  <div>
    <div
      className="period-info"
    >
      <p>
        YY/MM
      </p>
    </div>
    <ul
      className="period-filter"
    >
      <li>
        <button
          className=""
          onClick={[Function]}
          type="button"
        >
          Year
        </button>
      </li>
      <li>
        <button
          className=""
          onClick={[Function]}
          type="button"
        >
          Month
        </button>
      </li>
      <li>
        <button
          className=""
          onClick={[Function]}
          type="button"
        >
          Day
        </button>
      </li>
    </ul>
  </div>
  <div
    className="clear"
  />
</div>
`;
